server/utils/groupLoader.js

```diff
@@ -2,16 +2,84 @@ import groupService from '../services/group';
 
 function bulkScheduleEPRotation() {
   const schedulePromises = [];
-  groupService.getAllGroups()
-    .then((groups) => {
-      for (let i = 0; i < groups.length; i++) {
-        if (groups[i].escalationPolicy !== null) {
-          schedulePromises.push(groupService.scheduleEPRotation(groups[i]));
+  return new Promise((resolve, reject) => {
+    groupService.getAllGroups()
+      .then((groups) => {
+        for (let i = 0; i < groups.length; i++) {
+          if (groups[i].escalationPolicy !== null) {
+            schedulePromises.push(groupService.scheduleEPRotation(groups[i]));
+          }
         }
-      }
-      return Promise.all(schedulePromises);
-    })
-    .then(() => console.log('Group rotation has been scheduled'));
+        return Promise.all(schedulePromises);
+      })
+      .then((scheduledRotations, err) => {
```

Comment:
Would there ever be a second argument here? Does it not just bubble to the .catch()?